extensions/ffmpeg/src/main/java/com/google/android/exoplayer2/ext/ffmpeg/FfmpegDecoder.java

```diff
@@ -87,7 +87,11 @@ public FfmpegDecoderException decode(DecoderInputBuffer inputBuffer,
     }
     if (!hasOutputFormat) {
       channelCount = ffmpegGetChannelCount(nativeContext);
-      sampleRate = ffmpegGetSampleRate(nativeContext);
+      if ("alac".equals(codecName)) {
+        sampleRate = ByteBuffer.wrap(extraData, extraData.length - 4, 4).getInt();
```

Comment:
Is this equivalent to:
```
ParsableByteArray parsableExtraData = new ParsableByteArray(extraData);
parsableExtraData.setPosition(extraData.length - 4);
sampleRate = parsableExtraData.readUnsignedIntToInt();
```
It's IMO preferable to use that if so, since it checks the int really is unsigned.